commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -0,0 +1,161 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.android.volley.Cache;
+import com.android.volley.NetworkResponse;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.JsonRequest;
+import com.android.volley.toolbox.Volley;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+
+import java.io.UnsupportedEncodingException;
+
+public class MwVolleyApi {
+
+    private static RequestQueue REQUEST_QUEUE;
```

Comment:
This member is lazily initialized. As a client of this class, it's not clear to me that an initialization race condition may be started indirectly by calling request() which calls getQueue(). Consider adding `synchronized` to the getQueue() method or better yet, just initialize in the constructor and drop the Context member.